sb_vision/tokens.py

```diff
@@ -179,65 +37,68 @@ def __init__(self, id: int, certainty: float=0.0) -> None:
         """
         self.id = id
         self.certainty = certainty
+        self.pixel_coords = pixel_coords
+        self.pixel_centre = pixel_centre
 
     @classmethod
     def from_apriltag_detection(
-        cls,
-        apriltag_detection: 'ApriltagDetection',
-        image_size: Tuple[int, int],
-        distance_model: Optional[str],
+            cls,
+            apriltag_detection: 'ApriltagDetection',
+            camera_model: Optional[str],
     ) -> 'Token':
         """Construct a Token from an April Tag detection."""
+
+        pixel_coords = np.array([list(l) for l in apriltag_detection.p])
+
+        # centre of marker: average the corners
+        pixel_centre = tuple(
+            np.average(np.array([[1, 2], [1, 2], [1, 2]]), axis=0))
```

Comment:
Please can we make the wrapping here consistent with the rest of the repo's multi-line signatures? Essentially the pattern is that (if wrapping at all) an expression is either wrapped or not, and that its preceding and following items go on separate lines. The aim is to minimise the diff if another line/argument/expression/etc. needs to be added.

Here that means you _could_ wrap this like this:
``` python
        pixel_centre = tuple(
            np.average(np.array([[1, 2], [1, 2], [1, 2]]), axis=0),
        )
```
though the following would be nicer:
``` python
        pixel_centre = tuple(np.average(
            np.array([[1, 2], [1, 2], [1, 2]]),
            axis=0,
        ))
```
You could also do:
``` python
    pixel_centre = tuple(np.average(np.array([
        [1, 2],
        [1, 2],
        [1, 2]
    ]), axis=0))
```
though that's less clear about where the `axis=0` applies; if you like that list wrapping, you could combine both:
``` python
        pixel_centre = tuple(np.average(
            np.array([
                [1, 2],
                [1, 2],
                [1, 2]
            ]),
            axis=0,
        ))
```

Reply:
Thanks for pointing this out, also fun fact: that code is actually a bug in itself, it shouldn't statically return the average of 1 and 2, it should return the average of the corner pixels

Reply:
The decision to wrap this way is not conscious, it's just a default in pycharm